06_puppy/sirigithub/syncstore.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+type SyncStore struct {
+	sync.Map
+	sync.Mutex
+	currID int
+}
+
+func NewSyncStore() *SyncStore {
+	return &SyncStore{}
+}
+
+// CreatePuppy takes a user provided puppy and creates a new Puppy in the store
+// returns the ID of the new Puppy.
+func (m *SyncStore) CreatePuppy(puppy *Puppy) int {
+	m.Lock()
+	defer m.Unlock()
+	puppy.ID = m.currID
```

Comment:
Same as `MapStore`, because you are mutating the value passed in, that should be mentioned in the doc string.

Reply:
good point noted and updated.